cmake/init-compilation-flags.cmake

```diff
@@ -106,7 +106,8 @@ endif()
 add_compile_options(-Werror -Wall -Wextra -Wunused-function -Wfloat-conversion -Wno-sign-compare
                     -Wuninitialized -Wno-redundant-move -Wno-missing-field-initializers)
 if(COMPILE_RUNTIME_LIGHT)
-    add_compile_options(-Wno-vla-cxx-extension)
+    add_compile_options(-Wno-type-limits -Wno-attributes -Wno-ignored-attributes)
```

Comment:
Are this options necessary? Which compiler doesn't compile without them?

Reply:
It might be better to fix the places in the code that cause the errors

Reply:
ok

Reply:
I see that you add if for GCC in this place. In my opinion this check should't exist, but it's a different task